src/transaction.ts

```diff
@@ -264,10 +266,26 @@ const Transaction: any = {
     }
     await Request.postAuthorizationCode(config, authorization, transaction)
       .then((response: any) => {
-        validatesNonce(transaction, response['data']['nonce'])
-        accessResult = parseTokensAndStoreRefresh(config, response, transaction, { withPKCE: true })
+        try {
+          validatesNonce(transaction, response['data']['nonce'])
+          accessResult = parseTokensAndStoreRefresh(config, response, transaction, { withPKCE: true })
+        } catch (error) {
+          console.error("error while parseTokensAndstoreRefersh")
+          console.error(error)
+          errors.push({
+            error: 'transaction parse tokens',
+            error_description: `${error}`,
+            http_response: error.response
+          })
+          accessResult = {
+            ...accessResult,
+            valid: false,
+            errors: errors
+          }
+        }
```

Comment:
@docJerem `parseTokensAndStoreRefresh` throws some validation errors when needed but we do not forward them !